DataFormats/Provenance/test/indexIntoFile2_t.cppunit.cc

```diff
@@ -169,10 +198,10 @@ void TestIndexIntoFile2::testAddEntryAndFixAndSort() {
   CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[3].orderPHIDRun() == 0);
   CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[4].orderPHIDRun() == 1);
   CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[5].orderPHIDRun() == 1);
-  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[6].orderPHIDRun() == 3);
-  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[7].orderPHIDRun() == 3);
-  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[8].orderPHIDRun() == 3);
-  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[9].orderPHIDRun() == 3);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[6].orderPHIDRun() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[7].orderPHIDRun() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[8].orderPHIDRun() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[9].orderPHIDRun() == 2);
```

Comment:
What causes these to become 2 from 3?

Reply:
In addEntry when we are building a new IndexIntoFile, we assign a value to orderPHIDRun that is only used to sort the entries. We want each assigned value to be larger than the previous one to get the desired sort order. It only matters that the values are increasing. Before concurrency, we created this value only when adding a run using addEntry and its TTree entry number was easily available and always increasing and we simply used that because it was easy. 

Now we use the size of the container that the value is stored in. When you create a new value it goes into the container and it gets bigger by 1. We could have always done this.

We had to change, because now that things are concurrent we might generate new values when adding an event or lumi. We order by first appearance. When an event or lumi is added using addEntry we do not have the run TTree entry number available so it is no longer easy to use.

And finally, in this particular case the second run has two TTree entries. So in the original version orderPHIDRun was 0, 1, and then 3 (using Run TTree entry number to set the value). Now, it is 0, 1, and then 2 (using container size which increments by 1 for each new run).